migrated the spotify connector to use dscc-gen.

diminishedprime opened the pull request.

Lucretiel commented on spotify/scripts/dscc-scripts.js:
```diff
@@ -0,0 +1,79 @@
+  try {
+    await main();
+  } catch (e) {
+    console.error(`${e}`);
```
This should just be `console.error(e)`

Lucretiel commented on spotify/scripts/dscc-scripts.js:
```diff
@@ -0,0 +1,79 @@
+const main = async () => {
+  switch (args.script) {
+    case 'try_production': {
+      return await openDeployment(
```
You can replace all of these `return await` with just `return`

diminishedprime replied:
New code is in https://github.com/googledatastudio/dscc-scripts/blob/master/src/connector.ts and removed awaits from the switch block.

Lucretiel commented on spotify/scripts/dscc-scripts.js:
```diff
@@ -0,0 +1,79 @@
+  try {
+    await main();
+  } catch (e) {
+    console.error(`${e}`);
```
This should just be `console.error(e)`

diminishedprime replied:
I'll update that in the template.

Lucretiel commented on spotify/scripts/dscc-scripts.js:
```diff
@@ -0,0 +1,79 @@
+  return open(deploymentUrl);
+};
+
+const updateDeployment = async (deploymentId) => {
```
Does this function need to be async? Could it just be `(deploymentId) => exec(...)`

Lucretiel commented on spotify/scripts/dscc-scripts.js:
```diff
@@ -0,0 +1,79 @@
+
+const exec = (command, options) => {
+  return new Promise((resolve, reject) => {
+    const child = cp.exec(command, options, (err, stdout, stderr) => {
```
`exec` is a serious code smell, as it's a common source of injection attacks. Even if you think it's okay in this particular case, you should strongly prefer `execFile` unless you have a *really* compelling reason not to.

Lucretiel replied:
In particular, this function should have this interface:

    await exec(['npx', '@google/clasp', 'deploy', ...], {options}).then(...)

diminishedprime replied:
The new version uses execa, which is using execFile as suggested here.

Lucretiel commented on spotify/scripts/dscc-scripts.js:
```diff
@@ -0,0 +1,79 @@
+
+parser.addArgument(['-s', '--script'], {
+  choices: ['try_production', 'update_production', 'try_latest'],
+  dest: 'script',
```
I think `dest` is optional, it picks it out from `--script` if it uses the same behavior as Python argparse

Lucretiel commented on spotify/scripts/dscc-scripts.js:
```diff
@@ -0,0 +1,79 @@
+  required: true
+});
+
+const args = parser.parseArgs();
```
This should happen in `main`, not globally

diminishedprime replied:
This is fixed in the newer version.

diminishedprime commented:
Thanks for the review. After this (and other outstanding PRs) gets merged in, I'll switch to using @google/dscc-scripts as a dependency which addresses most of your feedback here.
